crux-lucene/test/crux/lucene_test.clj

```diff
@@ -238,6 +240,46 @@
         (catch Exception t
           (t/is (= "Lucene store latest tx mismatch" (ex-message (ex-cause t)))))))))
 
+(defn ^ICruxAPI start-rocks-lucene-node [{:keys [node-dir index-dir lucene-dir]}]
+  (c/start-node (merge {:crux/document-store {:kv-store {:crux/module `rocks/->kv-store,
+                                                         :db-dir (io/file node-dir "documents")}}
+                        :crux/tx-log {:kv-store {:crux/module `rocks/->kv-store,
+                                                 :db-dir (io/file node-dir "tx-log")}}}
+                       (when lucene-dir
+                         {:crux.lucene/lucene-store {:db-dir lucene-dir}})
+                       (when index-dir
+                         {:crux/index-store {:kv-store {:crux/module `rocks/->kv-store,
+                                                        :db-dir index-dir}}}))))
+
+(t/deftest test-lucene-node-restart
+  (t/testing "test restart with nil indexes"
+    (fix/with-tmp-dir "test-node" [node-dir]
+      (fix/with-tmp-dir "lucene" [lucene-dir]
+        (with-open [node (start-rocks-lucene-node {:node-dir node-dir :lucene-dir lucene-dir})]
+          (submit+await-tx node [[:crux.tx/put {:crux.db/id :ivan :name "Ivan"}]]))
+        (with-open [node (start-rocks-lucene-node {:node-dir node-dir :lucene-dir lucene-dir})]
+          (c/sync node)
+          (t/is (= (c/entity (c/db node) :ivan) {:crux.db/id :ivan :name "Ivan"}))))))
+
+  (t/testing "restart with partial indices (> lucene-tx-id tx-id)"
+    (fix/with-tmp-dir "test-node" [node-dir]
+      (fix/with-tmp-dir "lucene" [lucene-dir]
+        (fix/with-tmp-dir "indices" [index-dir]
+          (with-open [node (start-rocks-lucene-node {:node-dir node-dir :lucene-dir lucene-dir :index-dir index-dir})]
+            (submit+await-tx node [[:crux.tx/put {:crux.db/id :ivan :name "Ivan"}]]))
+          (with-open [node (start-rocks-lucene-node {:node-dir node-dir :lucene-dir lucene-dir})]
+            (submit+await-tx node [[:crux.tx/put {:crux.db/id :fred :name "Fred"}]]))
+          (with-open [node (start-rocks-lucene-node {:node-dir node-dir :lucene-dir lucene-dir :index-dir index-dir})]
+            (c/sync node)
+            (t/is (= (c/entity (c/db node) :fred) {:crux.db/id :fred :name "Fred"}))
```

Comment:
Can you re-use the `(c/db node)` here? It's something we try to convince users to do, and we do occasionally point them at our tests for model Crux usage :)